shopelectro/tests/tests_views.py

```diff
@@ -216,6 +216,47 @@ def test_pagination_step(self):
         response = self.get_category_page(query_string={'step': pagination_step})
         self.assertEqual(len(response.context['product_image_pairs']), pagination_step)
 
+    def test_pagination_404(self):
+        """Category page returns 404 for a nonexistent page number."""
+        self.assertEqual(
+            self.get_category_page(query_string={'page': 1000}).status_code,
+            404,
+        )
+
+    def assert_pagination_links(self, next, prev, page_number):
+        self.assertEqual(
+            get_page_number(self.client.get(next['href'])),
+            page_number + 1,
+        )
+        self.assertEqual(
+            get_page_number(self.client.get(prev['href'])),
+            page_number - 1,
+        )
+
+    def test_pagination_buttons(self):
+        """Buttons forward to a previous and a next pagination pages."""
+        page_number = 3
+        prev, next = BeautifulSoup(
+            self.get_category_page(query_string={'page': page_number}).content.decode('utf-8'),
+            'html.parser'
+        ).find(class_='catalog-pagination').find_all('a')
+
+        self.assert_pagination_links(next, prev, page_number)
```

Comment:
`self.assert_pagination_links(next, prev, page_number=3)` might be more good, then separated `page_number=3` var with it's state. Decide it